src/client/eppo-client.ts

```diff
@@ -517,33 +510,33 @@ export default class EppoClient implements IEppoClient {
   }
 
   private logAssignment(result: FlagEvaluation) {
+    const { flagKey, subjectKey, allocationKey, subjectAttributes, variation } = result;
```

Comment:
@schmit I think we can early exit here if an `allocationKey` is not present, that is a `noneResult` evaluation: https://github.com/Eppo-exp/js-client-sdk-common/blob/main/src/evaluator.ts#L76

That shouldn't even execute the `logAssignment` in that case but I understand since its nullable we need to play along with the ts compiler. semantically logging an assignment with `null` for the allocation and experiment is never correct.

Reply:
I see where you are coming from, but to push back:

suppose that there is some kind of unexpected issue (perhaps a wrongly configured targeting rule) that is discarding 10% of users and causing them to receive a null result. How would the user trace that down if we do not log the null results? With the logs it would be relatively easy to debug -- can query which subjects did not get assigned. 

On the other hand I also understand we might just be logging a lot of nulls and taking up space / cost

Reply:
on the spirit of not increasing the scope of this change, i'd propose we move this to a separate PR, if we decide to change the logic